parsec/core/fs/storage/chunk_storage.py

```diff
@@ -96,6 +96,44 @@ async def is_chunk(self, chunk_id: ChunkID) -> bool:
             manifest_row = cursor.fetchone()
         return bool(manifest_row)
 
+    async def get_local_chunk_ids(self, chunk_id: List[ChunkID]) -> List[ChunkID]:
+
+        local_chunk_ids = []
+        bytes_id_list = [(id.bytes,) for id in chunk_id]
+
+        async with self._open_cursor() as cursor:
+            # Can't use execute many with SELECT so we have to make a temporary table filled with the needed chunk_id
+            # and intersect it with the normal table
+            # create random name for the temporary table to avoid asynchronous errors
+            table_name = "temp" + secrets.token_hex(12)
+            table_name = "".join(chr for chr in table_name if chr.isalnum())
+
+            cursor.execute(f"""DROP TABLE IF EXISTS {table_name}""")
+            cursor.execute(
+                f"""CREATE TABLE IF NOT EXISTS {table_name}
+                    (chunk_id BLOB PRIMARY KEY NOT NULL -- UUID
+                );"""
+            )
+
+            cursor.executemany(
+                f"""INSERT OR REPLACE INTO
+            {table_name} (chunk_id)
+            VALUES (?)""",
+                iter(bytes_id_list),
+            )
+
+            cursor.execute(
+                f"""SELECT chunk_id FROM chunks INTERSECT SELECT chunk_id FROM {table_name}"""
+            )
+
+            manifest_rows = cursor.fetchall()
+            cursor.execute(f"""DROP TABLE IF EXISTS {table_name}""")
+        for row in manifest_rows:
+            index = bytes_id_list.index(row)
+            local_chunk_ids.append(chunk_id[index])
```

Comment:
This logic of looking back at the bytes seems a bit backward, I'd rather do:
```python
return [ChunkID(bytes=id_bytes) for (id_bytes,) in intersect_rows]
```
The `bytes` argument is not supported at the moment but this `UUID4` class can easily be modified as:
```python
class UUID4(UUID):
    __slots__ = ()

    def __init__(self, *args, **kwargs):
        if not args and not kwargs:
            super().__init__(bytes=uuid4().bytes)
        else:
            super().__init__(*args, **kwargs)
```